maven-core/src/main/java/org/apache/maven/project/MavenProject.java

```diff
@@ -1679,4 +1681,13 @@ public ProjectBuildingRequest getProjectBuildingRequest() {
     public void setProjectBuildingRequest(ProjectBuildingRequest projectBuildingRequest) {
         this.projectBuilderConfiguration = projectBuildingRequest;
     }
+
+    /**
+     * @since 4.0.0
+     * @return the rootdir for this project
+     * @throws IllegalArgumentException if the rootdir cannot be found
+     */
+    public Path getRootdir() {
+        return new DefaultRootLocator().locateMandatoryRoot(basedir != null ? basedir.toPath() : null);
+    }
```

Comment:
I think that's wrong.  We should not be lazy here and the use of `new DefaultRootLocator()` is wrong.